lib/puppet/type/ios_acl_entry.rb

```diff
@@ -0,0 +1,195 @@
+require 'puppet/resource_api'
+
+Puppet::ResourceApi.register_type(
+  name: 'ios_acl_entry',
+  docs: 'An entry for ACL',
+  features: ['remote_resource'],
+  attributes: {
+    name:         {
+      type:      'String',
+      desc:      'Name. Made up of access_list and the entry with a space seperator. eg. "list42 10" is from access_list list42 and entry 10.',
+      behaviour: :namevar,
+    },
+    ensure:      {
+      type:    'Enum[present, absent]',
+      desc:    'Whether this access list entry should be present or absent on the target system.',
+      default: 'present',
+    },
+    access_list:   {
+      type:    'String',
+      desc:    'Name of parent access list',
+    },
+    entry:    {
+      type:    'Integer',
+      desc:    'Name. Used as sequence number <1-2147483647>',
+    },
+    permission:      {
+      type:    'Enum["permit", "deny"]',
+      desc:    'Specify packets to forward/reject',
+    },
+    protocol:    {
+      type:    'Optional[Enum["ahp","eigrp","esp","gre","icmp","igmp","ip","ipinip","nos","ospf","pcp","pim","tcp","udp"]]',
```

Comment:
We may want to extend this to support an integer value as well.

```
Warning: Provider returned data that does not match the Type Schema for `ios_acl_entry[test 20]`
 Value type mismatch:
    * protocol: 8

cat-2960g#show ip access-lists         
Extended IP access list test
    10 permit udp any 0.0.0.2 255.255.255.0 log
    20 permit 8 host 1.1.1.1 any precedence routine tos max-reliability option traceroute log-input fragments
```

Reply:
Added a variant regex in the type to allow an integer protocol